src/mesh/mesh.cxx

```diff
@@ -322,3 +322,57 @@ ParallelTransform& Mesh::getParallelTransform() {
 Coordinates *Mesh::createDefaultCoordinates() {
   return new Coordinates(this);
 }
+
+
+Region<> & Mesh::getRegion3D(const std::string &region_name){
```

Comment:
why do you prefer strings over enums?
This changes a compile time check, to a run time check, and adds runtime overhead.

Reply:
My understanding is that an enum can't be extended (https://stackoverflow.com/a/1804924) and we may want to allow users/algorithms to register their own regions.

Reply:
An advantage is that it makes adding custom regions in user code easier. A user can add regions and refer to them with strings. 

An alternative might be to have a vector of regions (rather than a map): regions could then be referred to by an integer index. Adding a new region would return the index, probably wrapped in a class type rather than a raw integer. We could hard-wire built in regions to have particular indices perhaps. 

I'd like to see if this really is a significant overhead, before giving up the convenience of strings and maps.

Reply:
Ok, that makes sense.
I am not so much worried about the runtime cost, but having checks at compile time are awesome ...
Would it be viable to have both? Some hard coded ones as enum, and additional the interface with strings, such that users can define there own?
I don't like integers - they are not user friendly ...